src/Raven.Studio/typescript/common/shell/accessManager.ts

```diff
@@ -1,39 +1,121 @@
+                    return "Read Only access";
+            }
+        }
+        
+        return null;
+    }
+    
+    activeDatabaseAccessLevel = ko.pureComputed<Raven.Client.ServerWide.Operations.Certificates.DatabaseAccess>(() => {
+        const activeDatabase = activeDatabaseTracker.default.database();
+        if (activeDatabase) {
+            return this.getDatabaseAccessLevel(activeDatabase.name);
+        } 
+        
+        return null;
     });
     
+    isReadOnlyAccess = ko.pureComputed(() => this.activeDatabaseAccessLevel() === 'Read');
+    
+    canHandleOperation(requiredAccess: Raven.Client.ServerWide.Operations.Certificates.DatabaseAccess)  {
+        return ko.pureComputed(() => {
+            if (this.activeDatabaseAccessLevel() === 'Read') {
+                return false;
+            }
+
+            if (this.activeDatabaseAccessLevel() === 'ReadWrite' && requiredAccess === 'Admin') {
+                return false;
+            }
+
+            return true;
+        })
+    }
+    
     private createSecurityRule(enabledPredicate: KnockoutObservable<boolean>, requiredRoles: string) {
         return ko.pureComputed(() => {
             const enabled = enabledPredicate();
             const clearance = this.securityClearance();
+            
             if (enabled) {
                 return undefined;
-            } else {
-                return "Insufficient security clearance. <br /> Required: " + requiredRoles + "<br />Current: " + clearance;
             }
+            return this.getRuleHtml("Insufficient security clearance", requiredRoles, clearance);
         });
     }
 
-    disableIfNotClusterAdminOrClusterNode = this.createSecurityRule(this.clusterAdminOrClusterNode, "Cluster Admin");
+    private createDatabaseSecurityRule(requiredDatabaseAccess: Raven.Client.ServerWide.Operations.Certificates.DatabaseAccess) {
+        return ko.pureComputed(() => {
+            if (requiredDatabaseAccess === 'Admin' && this.operatorAndAbove()) {
+                return undefined;
+            }
+            
+            if (this.canHandleOperation(requiredDatabaseAccess)) {
+                return this.getRuleHtml("Insufficient database access", requiredDatabaseAccess, this.activeDatabaseAccessLevel());
+            }
+            
+            return undefined;
+        });
+    }
+    
+    private getRuleHtml(title: string, required: string, actual: string) {
+        return `<div class="text-left">
+                            <h4>${title}</h4>
+                            <ul>
+                                <li>Required: <strong>${required}</strong></li>
+                                <li>Actual: <strong>${actual}</strong></li>
+                            </ul>
+                        </div>`;
+    }
+
+    static activeDatabaseTracker = activeDatabaseTracker.default;
 
+    disableIfNotClusterAdminOrClusterNode = this.createSecurityRule(this.clusterAdminOrClusterNode, "Cluster Admin");
```

Comment:
can we follow existing naming strategry: disableIF...OrAbove?

Reply:
done